lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp

```diff
@@ -5956,13 +5961,11 @@ static void LegalizeDxilInputOutputs(Function *F,
       }
     } else if (qual == DxilParamInputQual::Inout) {
       // Only replace inout when (bLoad && bStore) == false.
-      bNeedTemp = true;
       bLoadOutputFromTemp = true;
       bStoreInputToTemp = true;
     }
 
     if (HLMatrixLower::IsMatrixType(Ty)) {
-      bNeedTemp = true;
```

Comment:
The bug was this when `qual == DxilParamInputQual::OutStream`. I removed the variable since it is redundant with the other two.